From 84b7d29d71a58e14c980b287a5795026faa6c108 Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@cloudera.com>
Date: Wed, 9 May 2012 11:53:52 -0700
Subject: [PATCH 1148/1357] MR1: CLOUDERA-BUILD. Fix MR1 TestFileSystem to work with FS serviceloader

  Reason: test is failing due to FS serviceloader changes
  Author: Alejandro Abdelnur
  Ref: CDH-5738
---
 src/test/org/apache/hadoop/fs/TestFileSystem.java |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/test/org/apache/hadoop/fs/TestFileSystem.java b/src/test/org/apache/hadoop/fs/TestFileSystem.java
index 3767a27..64c737d 100644
--- a/src/test/org/apache/hadoop/fs/TestFileSystem.java
+++ b/src/test/org/apache/hadoop/fs/TestFileSystem.java
@@ -668,7 +668,7 @@ public class TestFileSystem extends TestCase {
   @SuppressWarnings("unchecked")
   public <T extends TokenIdentifier> void testCacheForUgi() throws Exception {
     final Configuration conf = new Configuration();
-    conf.set("fs.cachedfile.impl", conf.get("fs.file.impl"));
+    conf.set("fs.cachedfile.impl", FileSystem.getFileSystemClass("file", conf).getName());
     UserGroupInformation ugiA = UserGroupInformation.createRemoteUser("foo");
     UserGroupInformation ugiB = UserGroupInformation.createRemoteUser("bar");
     FileSystem fsA = ugiA.doAs(new PrivilegedExceptionAction<FileSystem>() {
@@ -719,7 +719,7 @@ public class TestFileSystem extends TestCase {
   
   public void testCloseAllForUGI() throws Exception {
     final Configuration conf = new Configuration();
-    conf.set("fs.cachedfile.impl", conf.get("fs.file.impl"));
+    conf.set("fs.cachedfile.impl", FileSystem.getFileSystemClass("file", conf).getName());
     UserGroupInformation ugiA = UserGroupInformation.createRemoteUser("foo");
     FileSystem fsA = ugiA.doAs(new PrivilegedExceptionAction<FileSystem>() {
       public FileSystem run() throws Exception {
-- 
1.7.0.4

